packages/cms/pages/settings/roles-and-permissions/roles-and-permissions.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { FunctionComponent, useEffect, useState } from 'react'
+
+import { EuiBasicTable } from '@tensei/eui/lib/components/basic_table'
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter
+} from '@tensei/eui/lib/components/flyout'
+import { EuiTitle } from '@tensei/eui/lib/components/title'
+import { useGeneratedHtmlId } from '@tensei/eui/lib/services/accessibility'
+import { EuiButton } from '@tensei/eui/lib/components/button'
+import { EuiFieldText, EuiTextArea } from '@tensei/eui/lib/components/form'
+import { EuiText } from '@tensei/eui/lib/components/text'
+import { EuiFlexItem, EuiFlexGroup } from '@tensei/eui/lib/components/flex'
+import { EuiSwitch } from '@tensei/eui/lib/components/form/'
+import { EuiSpacer } from '@tensei/eui/lib/components/spacer/'
+import styled from 'styled-components'
+
+const RolesAndPermissionWrapper = styled.div`
+  padding: 20px 40px 0 40px;
+`
+const TableHeading = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding-left: 7px;
+`
+const AccordionWrapper = styled.div`
+  margin-bottom: 20px;
+`
+const AccordionHeader = styled.div`
+  height: 46px;
+  width: 100%;
+  padding: 16px 14px;
+  border: 1px solid #c9d3db;
+  background-color: #f5f7f9;
+  border-radius: 3px;
+`
+const Accordionbody = styled.div``
+const AccordionItem = styled.div`
+  padding: 20px 20px;
+  border-left: 0.4px solid #c9d3db;
+  border-right: 0.4px solid #c9d3db;
+  border-bottom: 0.4px solid #c9d3db;
+  height: 46px;
+`
+const Switch = () => {
+  const [checked, setChecked] = useState(true)
+
+  return (
+    <EuiSwitch
+      label=""
+      checked={checked}
+      onChange={() => setChecked(!checked)}
+    />
+  )
+}
+
+const FlyoutAccordion: React.FC = () => {
+  const [adminRoles, setAdminRoles] = useState<any[]>([])
+  useEffect(() => {
+    const fetchAdminRoles = async () => {
+      const [result] = await window.Tensei.api.get('admin-roles')
+      setAdminRoles(result?.data.data[0].adminPermissions)
```

Comment:
Seems like your method says `setAdminRoles` , but you are setting `adminPermissions` instead. Let our state be named after exactly what it does to ease maintenance in future. 

Secondly, from what i see, you are trying to get all permissions. We have an endpoint for that. `window.Tensei.api.get('admin-permissions')`